pymc4/model/base.py

```diff
@@ -66,12 +68,15 @@ def not_observed(var, *args, **kwargs):  # pylint: disable=unused-argument
             def get_mode(state, rv, *args, **kwargs):  # pylint: disable=unused-argument
                 return rv.distribution.mode()
             chain.insert(0, interceptors.Generic(after=get_mode))
-
-        with self.graph.as_default(), ed.interception(interceptors.Chain(*chain)):
+        tf.contrib.graph_editor.copy(self.graph, self.temp_graph)
+        # pylint: disable=not-context-manager
+        with self.temp_graph.as_default(), ed.interception(interceptors.Chain(*chain)):
```

Comment:
You reuse tmp graph created on step 1) suppose `cfg` has *any* tensor from original graph and you call your model with `temp_graph.as_default()`. That creates all variables for `temp_graph`. This looks fine unless the case I'm talking about. If you attempt to do an operation on tow variables from different graphs, you will get an exception
![image](https://user-images.githubusercontent.com/11705326/43569627-b72fffb2-9640-11e8-989a-c6067a946022.png)

Reply:
I do not know a solution for this case yet. I've seen some kind of working and relevant example in tensorflow codebase
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/framework/function.py#L347